### ISSUE TITLE
[pickers] How to trigger an event when clicking Cancel in the StaticDatePicker?

### ISSUE THREAD

**Issue by ermakov95** [mui/mui-x#8495]
![image](https://user-images.githubusercontent.com/89069249/229688030-26aba0e2-cc30-4815-84a8-224573dd692a.png)

**Comment by LukasTy** [comment]
Hello, @ermakov95.
Consider using the `onClose` prop in your case.
Clicking `cancel` resets the value to the last committed and calls the `onClose` callback.

**Comment by LukasTy** [comment]
> But onClose is deprecated and it have a second trigger - `onAccept` - when the value is not selected.
> Cancel is not unique trigger to onClose

You are correct about that. Basically, we lack a dedicated `onCancel` callback, which should also be endorsed for usage on Static pickers instead of `onClose`. 👌 

In the meantime, maybe you could create a [custom `ActionBar` component](https://mui.com/x/react-date-pickers/custom-layout/#css-customization) adding in your custom logic there? 🤔

**Comment by ermakov95** [comment]
> In the meantime, maybe you could create a [custom `ActionBar` component](https://mui.com/x/react-date-pickers/custom-layout/#css-customization) adding in your custom logic there?

I did the same

**Comment by LukasTy** [comment]
I'm re-opening this issue as it can be used to track the progress of implementing the addition of missing callback. 😉

**Comment by jake-lester** [comment]
Hello 👋 We noticed this is still an open issue affecting our component but It is unassigned. Are there any plans / code in-the-works to address this?

**Comment by nina-alin** [comment]
up !

**Comment by CHYGO1985** [comment]
up, I believe it is better to add a onCancel, since onClose is deprecated

**Comment by nina-alin** [comment]
Yeah, there isn't any props yet for telling the input what to do if the user clicks on the cancel button. I don't want to use a deprecated props as this will not be maintainable in the future. This button is basically unusable.

**Comment by michelengelen** [comment]
Status update: this is still an open gap. There is no dedicated `onCancel` callback today; `onClose` remains the only close-adjacent event, and it fires for every dismissal path, not just Cancel, which is the ambiguity raised in this thread. The workaround stands: override the `actionBar` slot and wire your own handler onto its Cancel button, as ermakov95 already did above.

Keeping this open as a real gap in the callback API, not a documentation issue.
